text/0156-columnar-encoding.md

```diff
@@ -0,0 +1,1018 @@
+
+In connection with these proposals, [we also propose corresponding improvements in the OpenTelemetry
+Collector-Contrib's `testbed` framework](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16835),
+in order to include OTel-Arrow in standard regression testing of the Collector.
+
+## Open Questions
+
+### Extending into other parts of the Arrow ecosystem
+
+A SQL support for telemetry data processing remains an open question in the current Go collector. The main OTelArrow query
+engine [Datafusion](https://github.com/apache/arrow-datafusion) is implemented in Rust. Several solutions can be
+considered: 1) create a Go wrapper on top of Datafusion, 2) implement a Rust collector dedicated to the end-to-end
+support of OTel Arrow, 3) implement a SQL/Arrow engine in Go (big project). A proof of concept using Datafusion has been
+implemented in Rust and has shown very good results.
+
+We believe that because the Arrow IPC mechanism and data format is intended for zero-copy use, we believe it is possible
+to use Arrow libraries written in other languages, for example within the Golang-based OpenTelemetry Collector.
+
+### Choosing row-oriented transport when it is more efficient
+
+The columnar representation is more efficient for transporting large homogeneous batches. The support of a mixed approach
+combining automatically column-oriented and row-oriented batches would allow to cover all scenarios. The development of
+a strategy to automatically select the best data representation mode is an open question.
+
+### Unary gRPC OTel Arrow and HTTP OTel Arrow
+
+The design currently calls for the use of gRPC streams to benefit from OTel Arrow transport.  We believe that some of
+this benefit can be had even for unary gRPC and HTTP requests with large request batches to amortize sending of
+dictionary and schema information.  This remains an area for study.
+
+## Future possibilities
+
+### Further-integrated compression techniques
+
+ZSTD offers a training mode, which can be used to tune the algorithm for a selected type of data. The result of this
+training is a dictionary that can be used to compress the data. Using this [dictionary](http://facebook.github.io/zstd/#small-data)
+can dramatically improve the compression rate for small batches. This future development will build on both the gRPC
+stream approach used in this proposal and the ability to send a ZSTD dictionary over the OTel Arrow stateful protocol,
+allowing us to train the ZSTD algorithm on the first batches and then update the configuration of the ZSTD
+encoder/decoder with an optimized dictionary.
+
+More advanced lightweight compression algorithms on a per column basis could be integrated to the OTel Arrow
+protocol (e.g. delta delta encoding for numerical columns)
```

Comment:
I'm not familiar with Arrow, is "delta delta encoding" the right term? (I've seen "delta encoding" before but not "delta delta").

Reply:
Delta-encoding and delta-delta encoding are related but in fact 2 distinct encoding. I will add a link.

Reply:
Thanks! This is not a blocker (totally due to my lack of knowledge in this space).